crates/ntx-builder/src/transaction.rs

```diff
@@ -70,19 +70,45 @@ type NtxResult<T> = Result<T, NtxError>;
 /// Provides the context for execution [network transaction candidates](TransactionCandidate).
 #[derive(Clone)]
 pub struct NtxContext {
-    pub block_producer: BlockProducerClient,
+    block_producer: BlockProducerClient,
 
     /// The prover to delegate proofs to.
     ///
     /// Defaults to local proving if unset. This should be avoided in production as this is
     /// computationally intensive.
-    pub prover: Option<RemoteTransactionProver>,
+    prover: Option<RemoteTransactionProver>,
 
     /// The store client for retrieving note scripts.
-    pub store: StoreClient,
+    store: StoreClient,
+
+    /// LRU cache for storing retrieved note scripts to avoid repeated store calls.
+    script_cache: Arc<Mutex<LruCache<Word, NoteScript>>>,
 }
 
 impl NtxContext {
+    /// Default cache size for note scripts.
+    ///
+    /// Each cached script contains the deserialized `NoteScript` object, so the actual memory usage
+    /// depends on the complexity of the scripts being cached.
+    const DEFAULT_SCRIPT_CACHE_SIZE: usize = 1000;
```

Comment:
I believe you can move the non-zero check here to compile time


```suggestion
    const DEFAULT_SCRIPT_CACHE_SIZE: NonZeroUsize = NonZeroUsize::new(1_000).unwrap();
```